registrants/src/UI/embc-registrant/src/app/feature-components/needs-assessment/needs-assessment-mapping.service.ts

```diff
@@ -233,26 +220,26 @@ export class NeedsAssessmentMappingService {
   private isSameAddress(evacAddress: RegAddress): string {
     const userPersonalAddress = this.profileDataService.primaryAddressDetails;
 
-    if (evacAddress === userPersonalAddress) {
+    if (JSON.stringify(evacAddress) === JSON.stringify(userPersonalAddress)) {
```

Comment:
object comparison with stringify will not always return correct results. You can use something like: compareObjects in profile.model.ts